In [20]:
import os
from dotenv import load_dotenv

# Load the .env file
load_dotenv()

# Access your API key
api_key = os.getenv("OPENWEATHER_API_KEY")

print("Your API Key is loaded:", api_key[:8] + "********")

# Now you can use the api_key variable in your application logic    
latitude = -1.286389
longitude = 36.817223

Your API Key is loaded: de8f1a0d********


In [ ]:
#Now let us have the API make a call to OpenWeatherMap to get the weather data for Nairobi, Kenya
url= "https://api.openweathermap.org/data/2.5/forecast"

params = {
    "lat": latitude,
    "lon": longitude,
    "exclude": "minutely,hourly,alerts", 
    "units": "metric",  
    "appid": api_key
}


In [25]:
# Make the API call
import requests
response = requests.get(url, params=params)
print("Status code:", response.status_code)

Status code: 401


401 error (Unauthorized) means OpenWeather is rejecting your request because the API endpoint or key doesn’t match the plan.